### ISSUE TITLE
Persisting unobserved computed values

### ISSUE THREAD

**Issue by adufilie** [mobxjs/mobx#875]
I have a use case (with mobx-react) where one render may use a particular computed value, the next may not, but a third render may use it again and we do not want it recomputed from scratch if no dependencies have changed.

If there is strong reason to keep the current behavior where it throws away the computed value as it becomes unobserved, can we please at least add an option for persisting computed values even when they are unobserved temporarily?

This is the same feature hinted at [in #706](https://github.com/mobxjs/mobx/issues/706#issuecomment-273191617) and my use case is related to #668 and #872 where I have a computed promise that I don't want to recompute.

In other words, I would like to eliminate the last part of the implementation behavior described in computedvalue.ts:  
> * If at any point it's outside batch and it isn't observed: reset everything and go to 1.

**Comment by mattruby** [comment]
See mobx-utils
https://github.com/mobxjs/mobx-utils/blob/master/README.md#keepalive
keepAlive will do what you want.

**Comment by adufilie** [comment]
I don't want to have to manually dispose the keepAlive, though, and I'm not sure that using keepAlive would still allow for automatic garbage collection.

The following line of code seems to be my issue, but removing it would cause existing assertions to fail and I'm not sure if other functionality implementations rely on this existing behavior:  https://github.com/mobxjs/mobx/blob/bc4dd19/src/core/derivation.ts#L199

**Comment by adufilie** [comment]
These lines, too:
https://github.com/mobxjs/mobx/blob/bc4dd19/src/core/derivation.ts#L73
https://github.com/mobxjs/mobx/blob/bc4dd19/src/core/computedvalue.ts#L78

**Comment by mattruby** [comment]
I think the keep alive computeds will still be cleaned up the same way.
They will just always compute. Give it a shot, let me know if I'm wrong.

**Comment by adufilie** [comment]
After thinking about it more I think you're right that they should be cleaned up as long as the cached value doesn't prevent it. I don't like the syntax though, so I will create my own decorator. Thanks.
